src/util/get_base_name.cpp

```diff
@@ -8,31 +8,39 @@ Author: CM Wintersteiger
 
 \*******************************************************************/
 
-
 #include "get_base_name.h"
 
 /// cleans a filename from path and extension
 /// \par parameters: a string
 /// \return a new string
 std::string get_base_name(const std::string &in, bool strip_suffix)
 {
-  size_t r=std::string::npos;
-  if(strip_suffix)
-    r=in.rfind('.', in.length()-1);
-  if(r==std::string::npos)
-    r=in.length();
+#ifdef _WIN32
+  // Windows now allows both '/' and '\\'
+  const std::size_t slash_pos = in.find_last_of("\\/");
+#else
+  const std::size_t slash_pos = in.rfind('/');
+#endif
 
-  size_t f=in.rfind('/', in.length()-1);
-  if(f==std::string::npos)
-    f=0;
+  std::size_t start_pos;
 
-  size_t fw=in.rfind('\\', in.length()-1);
-  if(fw==std::string::npos)
-    fw=0;
+  if(slash_pos == std::string::npos)
+    start_pos = 0;
+  else
+    start_pos = slash_pos + 1;
 
-  f = (fw>f)?fw:f;
+  std::size_t char_count;
 
-  if(in[f]=='/' || in[f]=='\\')
-    f++;
-  return in.substr(f, r-f);
+  if(strip_suffix)
+  {
+    std::size_t dot_pos = in.rfind('.');
+    if(dot_pos == std::string::npos || dot_pos < start_pos)
+      char_count = std::string::npos;
+    else
+      char_count = dot_pos - start_pos;
+  }
+  else
+    char_count = std::string::npos;
```

Comment:
I think a few lines of code can be saved by initialising `char_count` to `std::string::npos` and only updating if `(dot_pos != std::string::npos && dot_pos >= start_pos)`